packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol

```diff
@@ -34,6 +34,11 @@ contract L2OutputOracle is Initializable, Semver {
      */
     address public immutable PROPOSER;
 
+    /**
+     * @notice Minimum time (in seconds) that must elapse before a withdrawal can be finalized.
+     */
+    uint256 public immutable FINALIZATION_PERIOD_SECONDS;
```

Comment:
Is my understanding correct that this file pretty much the only change we made and every other change in this pr is just a cascading fix based on this change?

Reply:
Yeah- the only new code is the check for whether or not the first output that is deleted is finalized or not. Everything else is just moving `FINALIZATION_PERIOD_SECONDS` from the `OptimismPortal` -> `L2OutputOracle` so that it is in the proper spot for future changes. Since the `L2OutputOracle` needed to know about the finalization period, we just did it in this change.